lib/client.js

```diff
@@ -150,10 +150,36 @@ class Client {
       method  : 'GET',
       headers : {
         // https://tools.ietf.org/html/rfc6455
-        Connection             : 'Upgrade',
-        Upgrade                : 'WebSocket',
-        'Sec-WebSocket-Key'    : crypto.createHash('SHA1').digest('base64'),
-        'Sec-WebSocket-Version': 13
+        Connection               : 'Upgrade',
+        Upgrade                  : 'WebSocket',
+        'Sec-WebSocket-Protocol' : 'binary.k8s.io',
+        'Sec-WebSocket-Key'      : crypto.createHash('SHA1').digest('base64'),
+        'Sec-WebSocket-Version'  : 13
+      }
+    }, this.master_api);
+  }
+
+  exec(namespace, pod, container, termName) {
```

Comment:
Query parameters are best place in an option argument like in the `follow_log` method as there are optional.

Reply:
To be more generic, I would add the `command` argument instead of the `termName` option with a sensible default and let the caller pass the array with the correct `TERM` variable.

Reply:
Done in 01c403bf2d65d0bd93f6175b552c9d9f560de2c7.